src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -749,26 +749,34 @@ public override async Task<VersionStamp> GetTopLevelChangeTextVersionAsync(Cance
         {
             // We need to work out path to this document. Documents may not have a "real" file path if they're something created
             // as a part of a code action, but haven't been written to disk yet.
-            string effectiveFilePath;
+            string? effectiveFilePath = null;
 
             if (FilePath != null)
             {
                 effectiveFilePath = FilePath;
             }
             else if (Name != null && projectFilePath != null)
             {
-                effectiveFilePath = PathUtilities.CombinePathsUnchecked(PathUtilities.GetDirectoryName(projectFilePath), Name);
+                var projectPath = PathUtilities.GetDirectoryName(projectFilePath);
+
+                if (!RoslynString.IsNullOrEmpty(projectPath))
+                {
+                    effectiveFilePath = PathUtilities.CombinePathsUnchecked(PathUtilities.GetDirectoryName(projectFilePath), Name);
+                }
+            }
+
+            if (effectiveFilePath != null)
```

Comment:
❓ where are we at on the `!= null` vs `is object` argument? 

This is correct here imo just because it follows existing usage, but for my own sanity and future code what should we use?

Reply:
Oh, right, forgot about that. I'm decidedly in the "I have other hills to die on" camp so I'm not the one to ask about this.